schema/derive/src/lib.rs

```diff
@@ -27,7 +27,20 @@ pub fn schema_derive(input: TokenStream) -> TokenStream {
             .push(parse_quote! {iroha_schema::IntoSchema})
     });
 
-    impl_schema(&input).into()
+    let mut output: TokenStream = impl_schema(&input).into();
+
+    let input_string = format!("{}", input.to_token_stream());
+    if input_string.contains("usize") {
+        let to_add_tokens: TokenStream =
+            quote!(compile_error!("You may not use usize with IntoSchema");).into();
+        output.extend(to_add_tokens);
+    }
+    if input_string.contains("isize") {
+        let to_add_tokens: TokenStream =
+            quote!(compile_error!("You may not use isize with IntoSchema");).into();
+        output.extend(to_add_tokens);
+    }
```

Comment:
In procedural macros you have to panic if something is wrong. No need to insert `compile_error!` inside the output code.

From [procedural macros reference](https://doc.rust-lang.org/reference/procedural-macros.html):
>  Panics are caught by the compiler and are turned into a compiler error

Reply:
Also It seems to be a little bit hacky to scan input for `usize` or `isize` words such that. I think we should see, what @mversic thinks

Reply:
This is the best way to do it with Macros. However, Macros are not the best tool for the job. 

The real solution to this problem is a trait conflict that prevents users from implementing `IntoSchema` for a type `usize` altogether. 

So the macro either stays as is, and we then add the trait conflict, or the macro does a run-time check with proper type_id.

Reply:
I will open a new ticket for this, instead.

Reply:
#2004